data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/MigratingEndpointAdapter.java

```diff
@@ -35,4 +42,18 @@ protected void adapt(final Context ctx) throws Exception {
     final RestApiRequest request = new RestApiRequest(ctx, getMetadata());
     handleRequest(request);
   }
+
+  protected void handleStateByIdRequest(
```

Comment:
Not a fan of putting this in `MigratingEndpointAdapter` since we should be able to remove it when the migration is complete.

I do really like how we are likely to be able to write a heap of endpoints by just serialising this returned state differently. Really neat.

So maybe we just need to put this method in a subclass of `MigratingEndpointAdapter` that all the endpoints that return data just from the state extend, and this method could be the `handleRequest` implementation, the subclass just provides the `EndpointMetadata` which defines the serialisation.

Reply:
yep that'd be the other option, was just thinking this would get most of the logic common at least... it could equally be in a utility class, doesnt have to be on a handler, just makes it pretty... will have a play